src/GraphQL/Introspection/ISchemaComparer.cs

```diff
@@ -0,0 +1,134 @@
+using System.Collections.Generic;
+using GraphQL.Types;
+
+namespace GraphQL.Introspection
+{
+    /// <summary>
+    /// Provides the ability to order the schema elements upon introspection.
+    /// </summary>
+    public interface ISchemaComparer
+    {
+        /// <summary>
+        /// Returns a comparer for GraphQL types.
+        /// If this returns <see langword="null"/> then the original type ordering is preserved.
+        /// </summary>
+        IComparer<IGraphType> TypeComparer { get; }
+
+        /// <summary>
+        /// Returns a comparer for fields withing enclosing type.
+        /// If this returns <see langword="null"/> then the original field ordering is preserved.
+        /// </summary>
+        /// <param name="parent"> Parent type to which the fields belong. </param>
+        IComparer<IFieldType> FieldComparer(IGraphType parent);
+
+        /// <summary>
+        /// Returns a comparer for field arguments.
+        /// If this returns <see langword="null"/> then the original argument ordering is preserved.
+        /// </summary>
+        /// <param name="field"> The field to which the arguments belong. </param>
+        IComparer<QueryArgument> ArgumentComparer(IFieldType field);
+
+        /// <summary>
+        /// Returns a comparer for enum values.
+        /// If this returns <see langword="null"/> then the original enum value ordering is preserved.
+        /// </summary>
+        /// <param name="parent"> The enumeration to which the enum values belong. </param>
+        IComparer<EnumValueDefinition> EnumValueComparer(EnumerationGraphType parent);
+
+        /// <summary>
+        /// Returns a comparer for GraphQL directives.
+        /// If this returns <see langword="null"/> then the original directive ordering is preserved.
+        /// </summary>
+        IComparer<DirectiveGraphType> DirectiveComparer { get; }
+    }
+
+    /// <summary>
+    /// Default schema comparer. By default all elements are returned as is, no sorting is applied.
+    /// </summary>
+    public class DefaultSchemaComparer : ISchemaComparer
+    {
+        /// <inheritdoc/>
+        public virtual IComparer<IGraphType> TypeComparer => null;
+
+        /// <inheritdoc/>
+        public virtual IComparer<DirectiveGraphType> DirectiveComparer => null;
+
+        /// <inheritdoc/>
+        public virtual IComparer<QueryArgument> ArgumentComparer(IFieldType field) => null;
+
+        /// <inheritdoc/>
+        public virtual IComparer<EnumValueDefinition> EnumValueComparer(EnumerationGraphType parent) => null;
+
+        /// <inheritdoc/>
+        public virtual IComparer<IFieldType> FieldComparer(IGraphType parent) => null;
+    }
+
+    /// <summary>
+    /// Default schema comparer for GraphQL.NET v3.x.x.
+    /// By default only fields are ordered by their names within enclosing type.
+    /// </summary>
+    public sealed class LegacyV3SchemaComparer : DefaultSchemaComparer
```

Comment:
Do we need this class?

Reply:
Can we just include the source code for this class in the migration notes in case someone needs it?

Reply:
done